### PR TITLE
feat(proxy) implement delayed response with stream module

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Implements delayed responses with stream module that is used to short-circuit streams.

### CODE REVIEW

**Comment by locao** [comment]
`spec/02-integration/05-proxy/22-reports_spec.lua:383` is consistently failing now. Should the test be adapted to this new feature?

File: kong/init.lua [hunk 1]
```diff
@@ -281,6 +285,8 @@ local function execute_plugins_iterator(plugins_iterator, phase, ctx)
     if old_ws then
       ctx.workspace = old_ws
     end
+
+    ::continue::
```
Comment (javierguerragiraldez) [inline kong/init.lua:289]: this is a behaviour change: these two things (`reset_log` and resetting the workspace) are currently executed in the `delay_response and ctx.delayed_response` case, but with this patch they're skipped.  is that a desired change?

File: kong/pdk/response.lua [hunk 2]
```diff
@@ -861,21 +890,23 @@ local function new(self, major_version)
...
+      local ctx = ngx.ctx
+      ctx.KONG_EXITED = true
 
-        else
-          self.log.err("unable to proxy stream connection, " ..
-                       "status: " .. status .. ", err: ", body)
-        end
-      end
+      if ctx.delay_response and not ctx.delayed_response then
+        ngx.log(ngx.ERR, "STREAM DELAYED")
```
Comment (javierguerragiraldez) [inline kong/pdk/response.lua:897]: is this a real error, or a debug leftover?

Comment (bungle) [reply kong/pdk/response.lua:897]: Yes, a left-over, thank you!

**Review by dndx — APPROVED** [review]

**Pull request merged**